### PR TITLE
Fix update integration test

### PR DESCRIPTION
**Pull request by MDrakos**
https://www.pivotaltracker.com/story/show/178966184

### CODE REVIEW

**Comment by MDrakos** [comment]
There seems to be quite a bit of overlap with https://github.com/ActiveState/cli/pull/1465. @mdrohmann we should settle on which changes we would like to keep between the two.

**Comment by mdrohmann** [comment]
> There seems to be quite a bit of overlap with #1465. @mdrohmann we should settle on which changes we would like to keep between the two.

I suggest to use your PR for these fixes, but I think that you should cherry-pick this commit: https://github.com/ActiveState/cli/pull/1465/commits/7d26798eab86188ecebc4cdb1d40f5a069312f70 (It ensures the full log file is found on Windows everywhere.)  Does that sound good?  I could rebase my PR on top of yours afterwards, or just wait for you to merge the changes first.

File: test/integration/update_int_test.go [hunk 1]
```diff
@@ -276,14 +276,22 @@ func (suite *UpdateIntegrationTestSuite) TestUpdate() {
 			err := fileutils.Mkdir(fakeHome)
 			suite.Require().NoError(err)
 
+			before := fileutils.ListDir(ts.Dirs.Config, false)
```
Comment (mdrohmann) [inline test/integration/update_int_test.go:279]: Just suggest to use this instead: https://github.com/ActiveState/cli/pull/1465/commits/7d26798eab86188ecebc4cdb1d40f5a069312f70

**Review by mdrohmann — CHANGES_REQUESTED** [review]

**Review by Naatan — APPROVED** [review]

File: test/integration/update_int_test.go [hunk 2]
```diff
@@ -276,14 +276,22 @@ func (suite *UpdateIntegrationTestSuite) TestUpdate() {
...
 			cp.ExpectExitCode(0)
 
 			var logs string
 			if tt.TestUpdate {
-				logs = suite.pollForUpdateInBackground(cp.TrimmedSnapshot())
+				time.Sleep(500 * time.Millisecond)
+				after := fileutils.ListDir(ts.Dirs.Config, false)
+				onlyAfter, _ := funk.Difference(after, before)
+				logFile, ok := funk.FindString(onlyAfter.([]string), func(s string) bool { return strings.HasPrefix(filepath.Base(s), "state-installer") })
+				if ok {
```
[the remark this answers is not in the document]
Comment (MDrakos) [reply test/integration/update_int_test.go]: This has been moved after cherry-picking Martin's commit. It's not inside the `pollForUpdateInBackground` function and based on the `ok` we either return the log file contents or try again.

**Review by mdrohmann — APPROVED** [review]

**Review by Naatan — APPROVED** [review]

**Pull request merged**